net/src/net.cpp

```diff
@@ -89,6 +90,322 @@ int getAsync(lua_State* L)
     return lua_yield(L, 0);
 }
 
+static std::unordered_map<int, std::shared_ptr<uWS::App>> serverInstances;
+static std::unordered_map<int, std::shared_ptr<struct ServerLoopState>> serverStates;
```

Comment:
Can you use `DenseHashMap` from Luau's common library instead? It should be directly available already, and it is just strictly more performant. You'll need a tombstone value, but it seems like you're already starting at `1`, so `0` will work great.